### PR TITLE
refactor: replace PR _template_ with a reminder comment

### PR DESCRIPTION
**Pull request by Nemo157**
Simplifies filling out a PR description by just using the template as a checklist to verify against rather than attempting to use it to structure the description itself.

# Consent

Please register your opinion on the proposal in the following table:

| Dev              | +1 | +0 | -0 | -1 |
|------------------|:--:|:--:|:--:|:--:|
| @alexveecle      |    | x  |    |    |
| @HectorMRC       |    |✅ |    |    |
| @vE5li           |    |    |    |    |
| @arctic-alpaca   | ✅ |    |    |    |
| @vvvar           | ✅ |    |    |    |
| @Nemo157         | ✅ |    |    |    |
| @ForsakenHarmony |    |    |    |    |

<details>
    <summary>Legend</summary>

- `+1`: Agree
- `+0`: Positive no strong opinion
- `-0`: Negative no strong opinion
- `-1`: Concerns

When voting `Concerns`, please also explain any concerns in comments for discussion.

Approval requires all concerns to be resolved.

</details>

### CODE REVIEW

**Comment by Nemo157** [comment]
This is an alternative to my proposal to outright remove the template that I think strikes a better balance between reminding of the requirements while being less burdensome to actually fill out when opening a PR.

**Comment by Nemo157** [comment]
Another idea I had was to reduce the size by moving all but the first sentence out of the template into `CONTRIBUTING.md` (and then we could expand on the points a bit more comfortably there too), but we also have `veecle/nos/CONTRIBUTING.md` which I assume takes precedence if both exist.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,39 +1,35 @@
...
+
+ - PR title starts with an appropriate prefix as specified by [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#specification):
+
+       chore: [your title]
+       docs: [your title]
+       feat: [your title]
+       fix: [your title]
+       refactor: [your title]
+       rfc: [your title]
+       test: [your title]
```
Comment (arctic-alpaca) [inline .github/PULL_REQUEST_TEMPLATE.md:25]:
```suggestion
       refactor: [your title]
       test: [your title]
```

I think we can remove the `rfc` category.

**Comment by vE5li** [comment]
> For example, if you add a feature, don't include fixes you made along the way

Is this something we decided on? I remember the team having split opinions on this

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,39 +1,35 @@
...
+
+ - PR description explains what it intends to achieve (linking to issues where appropriate).
+
+ - For more complex PRs, PR description explains _how_ this was achieved.
+   Did you consider different approaches, were there any trade-offs you made?
+   More information about why you implemented your solution the way you did helps reviewers.
+
+(Feel free to delete this comment once you've checked your PR against the requirements, for a draft PR it can be useful to copy not-yet-completed steps out to a TODO section so reviewers know the current state).
```
Comment (arctic-alpaca) [inline .github/PULL_REQUEST_TEMPLATE.md:33]: Could you format the requirements as a TODO section to make copying more convenient?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,39 +1,35 @@
...
 
-## Intent
-<!---
-Please describe what you want to achieve here.
-Keep your PR limited to one logical change.
-For example, if you add a feature, don't include fixes you made along the way
-Please open another PR for more contributions.
--->
+ - PR is limited to a single logical change.
+   For example, if you add a feature, don't include fixes you made along the way
```
Comment (Nemo157) [inline .github/PULL_REQUEST_TEMPLATE.md:12]: > Is this something we decided on? I remember the team having split opinions on this

Yes, I think there was split opinion, but the template wasn't updated, this is just moving the existing checklist around. If someone remembers what the sort of consensus around it was it would be good to change it in another PR (I don't remember if it was to outright remove this, or just to weaken it in some way).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,39 +1,34 @@
...
+
+ - 🚧 PR title starts with an appropriate prefix as specified by [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#specification):
+
+       chore: [your title]
+       docs: [your title]
+       feat: [your title]
+       fix: [your title]
+       refactor: [your title]
+       test: [your title]
+
```
Comment (HectorMRC) [inline .github/PULL_REQUEST_TEMPLATE.md:25]: I would keep whatever regarding the PR title, body, description, etc at the very top.

This way it will be the first thing contributors will read when creating a PR.

In short i prefer listing requirements:
1. About PR title, description, linked issues, etc.
2. About commit messages.
3. About the PRs content (the code).

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,39 +1,34 @@
...
 
-chore: [your title]
-docs: [your title]
-feat: [your title]
-fix: [your title]
-refactor: [your title]
-rfc: [your title]
-test: [your title]
--->
+ - 🚧 Documentation in README, crate, module, function and/or other locations added/adapted.
```
Comment (HectorMRC) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: Aren't cargo-docs in crates, modules, functions etc mandatory to pass the CI?
Maybe we should list here only these requireme ts that are not covered by the CI.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -1,39 +1,34 @@
...
+ - 🚧 PR title starts with an appropriate prefix as specified by [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#specification):
+
+       chore: [your title]
+       docs: [your title]
+       feat: [your title]
+       fix: [your title]
+       refactor: [your title]
+       test: [your title]
+
+ - 🚧 PR description explains what it intends to achieve (linking to issues where appropriate).
```
Comment (HectorMRC) [inline .github/PULL_REQUEST_TEMPLATE.md:26]: What about spliting this in two bullet points?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -1,39 +1,34 @@
 <!---
+
 Thanks for creating a pull request!
 
-Please start your PR title with the appropriate prefix as specified by [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#specification):
+## TODO
```
Comment (HectorMRC) [inline .github/PULL_REQUEST_TEMPLATE.md:5]:
```suggestion
Please, make sure you addressed the following points before marking your PR as ready for review.
```

**Review by HectorMRC — CHANGES_REQUESTED** [review]
In general I agree with all the points we have here (I personally prefer checkboxes rather than bullet points, but i will not vote -1 for that XD).

**Comment by alexveecle** [comment]
+0 because this would be nicer on the NOS repo, as discussed in chat, but I could also have +1'ed.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (arctic-alpaca) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: CI cannot check whether some code-change requires a change to the documentation. 

At least for me, this can also be a helpful reminder to add stuff I might have forgotten about before creating a PR with CI immediately yelling at me.

Comment (Nemo157) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: We don't (yet) have semantic analyses in CI to determine whether the docs have been updated, or check any non-API documentation.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (Nemo157) [reply .github/PULL_REQUEST_TEMPLATE.md:5]: I made this a heading for @arctic-alpaca's point about making it easy to leave in uncompleted steps in a draft PR, so you can just delete the comment markers and the completed steps.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (HectorMRC) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Okay fair, so this point mostly stands for keeping documentation up-to-date rather than adding it (missing documentation do triggers a warning).

I believe the following form make the sentence simpler:
```suggestion
 - 🚧 Documentation is added and up-to-date in README, crate, module, function and/or other locations.
```

Comment (arctic-alpaca) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: For me personally, the current phrasing is easier to read.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (HectorMRC) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: Mmmmm okay, I didn't think about it like this.

The reason why I prefer the other way around is because PR related requirements are be more accessible when opening the PR. We can do it in the same page and while creating it. Then, as we go deeper in our requirements, we can mark it as a draft and then handle anything else (commits, doc, tests, etc). So it feels more natural to me.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (HectorMRC) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Then it's my bad English that triggers something in me when I read the action at the end XD

Feel free to mark this as resolved if all of you disagree on my changes

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (Nemo157) [reply .github/PULL_REQUEST_TEMPLATE.md:25]: (But I'm fine with any order if we get some more consensus on what makes sense to more people).

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (alexveecle) [reply .github/PULL_REQUEST_TEMPLATE.md]: Just today arctic-alpaca raised a point in the retro about small PRs taking a long time to merge, which I feel is very related to this.

If small PRs are not fast to complete, then there's a perverse incentive to either bundle them in a larger PR or not do the fix at all.

I would propose we keep this as-is... to force us to tackle the issue, although I believe there's a risk of not making small fixes they'd otherwise make :(

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (alexveecle) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: I don't have a strong opinion on this. But having enumerations in the middle of a sentence personally makes them hard to parse. You could simplify with:

```suggestion
 - 🚧 Updated documentation in README, crate, module, function and/or other locations.
```

, but I'm happy with any variant.

Comment (arctic-alpaca) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: This loses the `added` part.

Comment (Nemo157) [reply .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
 - 🚧 Added/updated relevant documentation in README, crate, module, function and/or other locations.
```

Comment (alexveecle) [reply .github/PULL_REQUEST_TEMPLATE.md:7]: Yes, I thought about this when proposing this. I had some thoughts about wording as  "complete documentation", "up to date documentation", "correct documentation", "adequate documentation", 
etc. but I don't think anyone will read this and say "hah! I don't have to add documentation for new stuff, the guidelines only talk about updating existing documentation!".

In general, I prefer conciseness over accuracy, because conciseness in general helps understanding, and sometimes accuracy is not worth the loss of conciseness.

**Review by HectorMRC — APPROVED** [review]

**Review by alexveecle — DISMISSED** [review]
Just a quick test, I'll dismiss this.

**Review by alexveecle — APPROVED** [review]
